packages/react-i18n/src/decorator.ts

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import {RegisterOptions} from './manager';
+import {useI18n} from './hooks';
+import I18n from './i18n';
+
+export interface WithI18nProps {
+  i18n: I18n;
+}
+
+export function withI18n(i18nOptions: RegisterOptions) {
+  return <P extends WithI18nProps>(
```

Comment:
This no longer does the `hoistNonReactStatics` stuff, which I think was part of the old one?

Reply:
I added `hoistStatics` which was what the old one used. https://github.com/Shopify/quilt/blob/master/packages/react-i18n/src/decorator.tsx#L153